src/components/Search/effects.ts

```diff
@@ -0,0 +1,25 @@
+import { call, put, throttle } from 'redux-saga/effects';
+
+import * as ActionType from './actionTypes';
+import * as Action from './actions';
+import { SEARCH_API_DELAY } from './constants';
+import { queryFeatures, FeatureQuery, Features } from '../../api';
+
+function* getFeatures(action: Action.SearchQuery) {
+    try {
+        const features: Features[] = yield call(
+            queryFeatures,
+            action.payload.query
+        );
+
+        yield put(Action.results(action.payload.field, features));
+    } catch (error) {
+        yield put(Action.error(action.payload.field, error));
+    }
+}
+
+function* getFeaturesSaga() {
+    yield throttle(SEARCH_API_DELAY, ActionType.QUERY, getFeatures);
```

Comment:
I guess this is supposed to be in place of the debounce, but it doesn't work. Typing in the search box fires off requests per letter typed.

Reply:
surprising... not what I observed. Will have to try to reproduce this. Have you got specific steps you used?

Reply:
Looks like it might be that the first letter always fires off regardless, then the throttle kicks in after

Reply:
I can't re-produce this. I've switched to a `debounce` to make the behaviour a little more expected. @dweemx can you re-produce?

Reply:
@MaybeJustJames No I can't reproduce this. I just tested your branch (9084c70cec3b6e50557ff20e2e905eaa5c9289e3) and I don't see multiple requests firing as I type a gene in the box